syntax = "proto3";

package stock_testing;

option java_package = "com.chi.ssetest.protos";
option java_outer_classname = "SetupConfig";

message MarketPermission {
	string Level = 1;
	string SseLevel = 2;
	string CffLevel = 3;
	string DceLevel = 4;
	string CzceLevel = 5;
	string FeLevel = 6;
	string GILevel = 7;
	string ShfeLevel = 8;
	string IneLevel = 9;
	repeated string HKPerms = 10;
}

message SDKConfig {
	string appKey = 1;
	map<string, string> serverSites = 2;
	MarketPermission marketPerm = 3;
}

message TestcaseConfig {
	string testcaseID = 1;
	bool continueWhenFailed = 2;
	int64 roundIntervalSec = 3;
	repeated string paramStrs = 4;
}

message RunnerConfig {
	string jobID = 1;
	string runnerID = 2;
	SDKConfig sdkConfig = 3;
	repeated TestcaseConfig casesConfig = 4;
}